include/parsers/qasm3_parser/Statement.hpp

```diff
@@ -0,0 +1,435 @@
+};
+
+class Constant : public Expression {
+private:
+  std::variant<int64_t, double> val;
+  bool isSigned;
+  bool isFp;
+
+public:
+  Constant(int64_t value, bool valueIsSigned)
+      : val(value), isSigned(valueIsSigned), isFp(false) {}
+
+  explicit Constant(double value) : val(value), isSigned(true), isFp(true) {}
+
+  [[nodiscard]] bool isInt() const { return !isFp; }
+  [[nodiscard]] bool isSInt() const { return !isFp && isSigned; }
+  [[nodiscard]] bool isUInt() const { return !isFp && !isSigned; }
+  [[nodiscard]] bool isFP() const { return isFp; };
+  [[nodiscard]] virtual int64_t getSInt() const { return std::get<0>(val); };
+  [[nodiscard]] virtual uint64_t getUInt() const {
+    return static_cast<uint64_t>(std::get<0>(val));
+  };
+  [[nodiscard]] virtual double getFP() const { return std::get<1>(val); };
+  [[nodiscard]] virtual double asFP() const {
+    if (isFp) {
+      return getFP();
+    }
+    if (isSigned) {
+      return static_cast<double>(getSInt());
+    }
+    return static_cast<double>(getUInt());
+  };
+
+  std::string getName() override { return "Constant"; }
+};
+
+class BinaryExpression : public Expression,
+                         public std::enable_shared_from_this<BinaryExpression> {
+public:
+  enum Op {
+    Power,
+    Add,
+    Subtract,
+    Multiply,
+    Divide,
+    Modulo,
+    LeftShift,
+    RightShift,
+    LessThan,
+    LessThanOrEqual,
+    GreaterThan,
+    GreaterThanOrEqual,
+    Equal,
+    NotEqual,
+    BitwiseAnd,
+    BitwiseXor,
+    BitwiseOr,
+    LogicalAnd,
+    LogicalOr,
+  };
+
+  Op op;
+  std::shared_ptr<Expression> lhs;
+  std::shared_ptr<Expression> rhs;
+
+  BinaryExpression(Op opcode, std::shared_ptr<Expression> l,
+                   std::shared_ptr<Expression> r)
+      : op(opcode), lhs(std::move(l)), rhs(std::move(r)) {}
+
+  std::string getName() override { return "BinaryExpr"; }
+};
+
+class UnaryExpression : public Expression,
+                        public std::enable_shared_from_this<UnaryExpression> {
+public:
+  enum Op {
+    BitwiseNot,
+    LogicalNot,
+    Negate,
+    DurationOf,
+    Sin,
+    Cos,
+    Tan,
+    Exp,
+    Ln,
+    Sqrt,
+  };
+
+  std::shared_ptr<Expression> operand;
+  Op op;
+
+  UnaryExpression(Op opcode, std::shared_ptr<Expression> expr)
+      : operand(std::move(expr)), op(opcode) {}
+
+  std::string getName() override { return "UnaryExpr"; }
+};
+
+class IdentifierExpression
+    : public Expression,
+      public std::enable_shared_from_this<IdentifierExpression> {
+public:
+  std::string identifier;
+
+  explicit IdentifierExpression(std::string id) : identifier(std::move(id)) {}
+
+  std::string getName() override {
+    return std::string{"IdentifierExpr ("} + identifier + ")";
+  }
+};
+
+class IdentifierList : public Expression,
+                       public std::enable_shared_from_this<IdentifierList> {
+public:
+  std::vector<std::shared_ptr<IdentifierExpression>> identifiers;
```

Comment:
```suggestion
  std::vector<std::shared_ptr<IdentifierExpression>> identifiers{};
```
fairly sure that should fix the clang-tidy warning.